src/core/hooks/useCopyBlockIds.ts

```diff
@@ -4,16 +4,26 @@ import { cutBlockIdsAtom } from "./useCutBlockIds";
 
 export const copiedBlockIdsAtom: any = atom<Array<string>>([]);
 
-export const useCopyBlockIds = (): [Array<string>, Function] => {
+export interface CopiedBlock {
+  id: string;
+  data: any;
+}
+
+export const useCopyBlockIds = (): [Array<string>, (blocks: Array<CopiedBlock>) => void] => {
   const [ids, setIds] = useAtom(copiedBlockIdsAtom);
   const resetCutBlockIds = useSetAtom(cutBlockIdsAtom);
 
   const setCopiedBlockIds = useCallback(
-    (blockIds: Array<string>) => {
+    (blocks: Array<CopiedBlock>) => {
+      const blockIds = blocks.map(block => block.id);
       setIds(blockIds);
       resetCutBlockIds([]);
+      
+      // Store blocks data in clipboard
+      navigator.clipboard.writeText(JSON.stringify(blocks));
```

Comment:
this is not handling nested blocks. its oonly copying the selected block

Reply:
Yes, Need to copy its children as well.